client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -226,7 +231,20 @@ public class ControllerImpl implements Controller {
     private final Supplier<Long> requestIdGenerator = RandomFactory.create()::nextLong;
 
     private final long timeoutMillis;
-    
+
+    /**
+     * Maximum time that segment endpoint details are held in the cache.
+     */
+    private final Duration segmentEntryExpirationTime = Duration.ofMinutes(5);
+    /**
+     * Maximum number of segment endpoint entries in the cache.
+     */
+    private final int maxCacheSize = 1000;
+
+    @Getter(value = AccessLevel.PACKAGE)
+    private final SimpleCache<Long, CachedPravegaNodeUri> endPointCacheMap;
```

Comment:
The key for this map should be `Segment`. Not the `id` of the segment. (The whole of the segment determines the mapping)

Reply:
Incorporated changes